### PR TITLE
Feature/datalake buffer

### PR DESCRIPTION
**Pull request by marianogaragiola**

### CODE REVIEW

File: splight_lib/client/datalake/remote_client.py [hunk 1]
```diff
@@ -43,6 +46,7 @@ def save(
     ) -> List[dict]:
         instances = instances if isinstance(instances, list) else [instances]
         url = self._base_url / f"{self._PREFIX}/save/"
+        collection = camelcase(collection)
```
Comment (matisilva) [inline splight_lib/client/datalake/remote_client.py:49]: miedo con el guion bajo

**Review by matisilva — APPROVED** [review]

File: splight_lib/client/datalake/remote_client.py [hunk 2]
```diff
@@ -262,3 +266,74 @@ def execute_query(
...
+class BufferedRemoteDatalakeClient(RemoteDatalakeClient):
+    _PREFIX = "v2/engine/datalake"
+
+    def __init__(
+        self,
+        base_url: str,
+        access_id: str,
+        secret_key: str,
+        buffer_size: int = 10,
+        buffer_timeout: float = 30,
```
Comment (francozacco) [inline splight_lib/client/datalake/remote_client.py:280]: I think we should use the same constants you defined in the settings here.

**Review by ignacioFernandez1 — APPROVED** [review]

File: splight_lib/client/datalake/remote_client.py [hunk 3]
```diff
@@ -262,3 +266,74 @@ def execute_query(
...
+            tags=LogTags.DATALAKE,
+        )
+
+    def save(
+        self, collection: str, instances: Union[List[Dict], Dict]
+    ) -> List[Dict]:
+        instances = instances if isinstance(instances, List) else [instances]
+        if collection not in self._data_buffers:
+            raise Exception("Invalid collection name")
+        self._data_buffers[collection].add_documents(instances)
```
Comment (francozacco) [inline splight_lib/client/datalake/remote_client.py:320]: At some point shouldn't we have to save the data to the datalake too?

File: splight_lib/models/component.py [hunk 4]
```diff
@@ -174,7 +174,8 @@ class ComponentObject(SplightObject):
 
 
 class RoutineEvaluation(SplightDatalakeBaseModel):
-    _collection_name = "routineEvaluations"
+    # _collection_name = "routineEvaluations"
```
Comment (francozacco) [inline splight_lib/models/component.py:177]: commented line...

File: splight_lib/client/datalake/remote_client.py (hunk 3 above)
Comment (marianogaragiola) [reply splight_lib/client/datalake/remote_client.py]: There is a thread that is responsible for sending the documents to the datalake

Comment (francozacco) [reply splight_lib/client/datalake/remote_client.py]: Oh yeah that's true!

**Review by francozacco — APPROVED** [review]

**Pull request merged**